gef.py

```diff
@@ -1226,14 +1229,15 @@ class GlibcArena:
     Ref: https://github.com/sploitfun/lsploits/blob/master/glibc/malloc/malloc.c#L1671"""
 
     def __init__(self, addr: str) -> None:
+        self.__arena: Union["gdb.Value", MallocStateStruct]
         try:
             arena = gdb.parse_and_eval(addr)
             malloc_state_t = cached_lookup_type("struct malloc_state")
-            self.__arena = arena.cast(malloc_state_t)
+            self.__arena = arena.cast(malloc_state_t)  # "gdb.Value"
```

Comment:
We can't do `self.__arena : "gdb.Value" = ...`?

Reply:
actually that is a comment for understanding the code better. I put it there to emphasize it for the reader because actually what we do is highly problematic (to sometimes have the arena be a `gdb.Value` and sometimes a `MallocStateStruct` with different ways of handling them)

Reply:
Ah. Let's reword it with basically what you just said?